src/main/java/github/workflow/kata/Grid.java

```diff
@@ -6,10 +6,69 @@
  * A gaming grid.
  */
 public class Grid {
-    private final String gridDef;
+    private String gridDef;
 
     public Grid(String gridDef) {
-        this.gridDef = gridDef;
+        // make sure the gridDef has no spaces, as equals does a string comparison, while the input can have spaces
+        // around the grid definition.
+        String niceDef = "";
+        String[] rows = gridDef.trim().split("\n");
+        for (String s : rows) {
+            char[] cols = s.trim().toCharArray();
+            for (char c : cols) {
+                niceDef += c;
+            }
+            // add a new-line
+            niceDef += '\n';
+        }
+        this.gridDef = niceDef;
+    }
+
+    public void nextGeneration() {
+        // so, lets try to implement the actual code.
+        // not so happy with the implementation, it's "a bit" dirty, but it works. Refactor this later.
+        String newDef = "";
+        String[] rows = gridDef.split("\n");
+        for (int row = 0; row < rows.length; row++) {
+            char[] cols = rows[row].toCharArray();
+            for (int col = 0; col < cols.length; col++) {
+                char ch = cols[col];

```

Comment:
```suggestion
        for (String row : gridDef.split("\n")) {
            for (char ch :  row.toCharArray()) {
```

Reply:
it looks differend after the refactorings I have done.